edtlib: binding: Add a title keyword

rruuaanng opened the pull request:
Add a 'title' keyword to the binding to provide a short description of the binding, while 'description' serves as the long description.

fixes: #86265

decsny commented:
@kartben I dont know how this plays with documentation generation for the bindings, can you help reviewing this PR regarding those implications for what would need done or not with the documentation

rruuaanng commented:
Perhaps we could consider concatenating the description of the `title` keyword to the `description`, meaning that the title remains visible to users only at the textual level, serving to enhance the readability of the text, while the document continues to be generated in the following format for the description:

doc
```
# Description
--------------------
title body

description body
...
--------------------
```

binding:
```
title: xxx

description: |
  xxx
  ...
```

rruuaanng commented:
https://builds.zephyrproject.io/zephyr/pr/86533/docs/build/dts/api-usage.html

rruuaanng commented:
Change log:

Add content merging of title and description.

mbolivar reviewed (changes requested):
What's the use case here? Is it for dtsh (and if so, is @dottspina on board with this change)?

This PR is missing an update to doc/build/dts/bindings-syntax.rst -- and I think that update should make it clear what the use case is and what the recommended uses are for the two fields, etc.

rruuaanng commented:
Oh @mbolivar, I saw your suggestion in the source issue, thanks for the reminder, I seem to have forgotten that `bindings-syntax.rst` exists.

mbolivar commented on doc/build/dts/bindings-syntax.rst:
```diff
@@ -19,6 +19,16 @@ like this:
+   # description |
+   #   A piece of content with 20 lines.
+   #   ...
+   title: Concise title for the long description [optional]
```
Thanks for the update. Please see the below text as well:

```
These keys are explained in the following sections.

.. _dt-bindings-description:

Description
***********
```

This is adding a key without adding a corresponding section; please fix that up too. Thanks!

rruuaanng committed:
edtlib: binding: Add a title keyword

Add a 'title' keyword to the binding to provide a short
description of the binding, while 'description' serves as
the long description.

Signed-off-by: James Roy <rruuaanng@outlook.com>
